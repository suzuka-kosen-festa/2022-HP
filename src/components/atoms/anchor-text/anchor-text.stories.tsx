import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { AnchorText } from ".";

type T = typeof AnchorText;
type Story = ComponentStoryObj<T>;

const data = {
  link: "#",
  role: "link",
  sentence: "This is test",
  tabindex: 0,
};

export default {
  args: { children: data.sentence, href: data.link, role: data.role, tabIndex: data.tabindex },
  component: AnchorText,
} as ComponentMeta<T>;

export const Default: Story = {};
