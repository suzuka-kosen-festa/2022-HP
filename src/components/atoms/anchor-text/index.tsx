import tw from "twin.macro";

const AnchorText = tw.a`font-zen text-xs sm:text-text text-base`;

export { AnchorText };
