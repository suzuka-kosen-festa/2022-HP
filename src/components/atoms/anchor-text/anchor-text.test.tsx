import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./anchor-text.stories";

const { Default } = composeStories(stories);

const options = { name: "This is test" };

describe("(components) atoms/anchor-text", () => {
  test("to be atoms", () => {
    const { container } = render(<Default />);
    expect(container).toBeAtom();
  });
  test("to be [role=link]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("link", options)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
