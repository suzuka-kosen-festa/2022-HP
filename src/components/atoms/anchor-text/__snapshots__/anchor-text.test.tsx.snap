// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) atoms/anchor take snap shot 1`] = `
<div>
  <a
    class="twin-1w1mmlr-AnchorText ey6ijqd0"
    href="#"
    role="link"
  >
    This is test
  </a>
</div>
`;
